compiler/rustc_hir_analysis/src/collect/type_of/opaque.rs

```diff
@@ -324,6 +324,9 @@ struct RpitConstraintChecker<'tcx> {
 impl RpitConstraintChecker<'_> {
     #[instrument(skip(self), level = "debug")]
     fn check(&self, def_id: LocalDefId) {
+        if self.found.references_error() {
+            return;
+        }
         // Use borrowck to get the type with unerased regions.
         let concrete_opaque_types = &self.tcx.mir_borrowck(def_id).concrete_opaque_types;
```

Comment:
I still don't think this is the right fix, because we should be able to call `mir_borrowck` on any def id at this point without causing issues..?

Either that, or we should check if `self.tcx.typeck(def_id).tainted_by_errors.is_some()` rather than `self.found.references_error()`...